core/chains/terra/terratxm/txm.go

```diff
@@ -154,7 +156,7 @@ func (txm *Txm) sendMsgBatch() {
 			// after it was added for this to happen. Retry on next poll should the key be re-added.
 			continue
 		}
-		txm.sendMsgBatchFromAddress(gp, sender, key, msgs)
+		txm.sendMsgBatchFromAddress(prices["uluna"], sender, key, msgs)
```

Comment:
What happens if `prices["uluna"]` is empty/zero/unset? Would it be worth checking up front before the loop, to short circuit with one clean error?

Reply:
that should be impossible/would be a programmer error but couldn't hurt